HIP/hip-513.md

```diff
@@ -0,0 +1,293 @@
+     */
+    System = 4;
+}
+
+/**
+ * A finder grained action with a function result. Sometimes called "internal transactions." The function call itself
+ * will be the first action in a list, followed by sub-action in the order they were executed.
+ */
+message ContractAction {
+
+    /**
+     * The type of this action.
+     */
+    ContractActionType callType = 1;
+
+    /**
+     * Only the first action can come from an account, the rest will come from contracts.  Because of DELEGATECALL
+     * and CALLCODE the caller of actions whose parent is an account may also be an account.
+     */
+    oneof caller {
+        /**
+         * If the caller was a regular account, the AccountID.
+         */
+        AccountID callingAccount = 2;
+
+        /**
+         * If the caller was a smart contract account, the ContractID.
+         */
+        ContractID callingContract = 3;
+    }
+
+    /**
+     * Gas given to this action, the upper limit of gas this action can spend.
+     */
+    uint64 gas = 4;
+
+    /**
+     * Bytes passed in as input data to this action.
+     */
+    bytes input = 5;
+
+    /**
+     * Who this action is directed to.
+     */
+    oneof recipient {
+        /**
+         * The AccountID of the recipient if the recipient is an account. Only HBars will be transferred, no other side
+         * effects should be expected.
+         */
+        AccountID recipientAccount = 6;
+
+        /**
+         * The ContractID of the recipient if the recipient is a smart contract.
+         */
+        ContractID recipientContract = 7;
+
+        /**
+         * If the action was directed at an invalid solidity address, what that address was.
+         */
+        bytes invalidSolidityAddress = 8;
+    }
+
+    /**
+     * The value (int tinybars) that is associated with this action.
+     */
+    int64 value = 9;
+
+    /**
+     * The actual gas spent by this action.
+     */
+    uint64 gasUsed = 10;
+
+    /**
+     * Was the action successful?
+     */
+    bool success = 11;
+
+    /**
+     * The result data of the action.
+     */
+    oneof resultData {
+
+        /**
+         * If successful, the output bytes of the action.
+         */
+        bytes output = 12;
+
+        /**
+         * The contract itself cause the transaction to fail via the `REVERT` operation
+         */
+        bytes revertReason = 13;
+
+        /**
+         * The transaction itself failed without an explicit `REVERT`
+         */
+        bytes error = 14;
+    }
+
+    /**
+     * The nesting depth of this call. The original action is at depth=0.
+     */
+    uint32 callDepth = 15;
+}
+```
+
+For every smart contract transaction a `ContractAction` record should be added to the sidecar record.
+
+### Add ContractBytecode sidecar record type
+
+Add and populate a new sidecar record of type `ContractBytecode`. It will
+have two fields - `deploymentBytecode` and `runtimeBytecode`. Fields will
```

Comment:
Will this finally allow child contract creates to send their init code and runtime bytecode? And if so the timestamp associated with that be will be for the child transaction, right? Can we call this scenario out here.

Reply:
DONE